src/plugins/visualizations/public/embeddable/visualize_embeddable.ts

```diff
@@ -245,26 +244,20 @@ export class VisualizeEmbeddable extends Embeddable<VisualizeInput, VisualizeOut
   hasInspector = () => Boolean(this.getInspectorAdapters());
 
   onContainerLoading = () => {
-    this.domNode.setAttribute('data-render-complete', 'false');
+    this.renderComplete.dispatchInProgress();
     this.updateOutput({ loading: true, error: undefined });
   };
 
-  onContainerRender = (count: number) => {
-    this.domNode.setAttribute('data-render-complete', 'true');
-    this.domNode.setAttribute('data-rendering-count', count.toString());
+  onContainerRender = () => {
+    this.renderComplete.dispatchComplete();
     this.updateOutput({ loading: false, error: undefined });
```

Comment:
do we still need this or could we make a component use this render complete stuff directly ?

Reply:
I'm not sure how `{ loading: false, error: undefined }` *output* is used, but I will see if I can remove/modify it.

Reply:
Yes, currently we still need this as it is used in multiple places and embeddable output is reactive but render-complete helper is not. I could make the render-complete helper reactive, too, then we could remove this from output and use the render complete directly. But let's do it in a separate PR.